### PR TITLE
Use TimerOutput::Scope in step-32.

### PR DESCRIPTION
**Pull request by bangerth**
This program still used the old, manual enter_section/leave_section()
functions. We know these days how to do this better.

### CODE REVIEW

File: examples/step-32/step-32.cc [hunk 1]
```diff
@@ -1972,10 +1972,23 @@ namespace Step32
   // locale (which we get using the constructor call
   // <code>std::locale("")</code>) implies printing numbers with a comma
   // separator for every third digit (i.e., thousands, millions, billions).
+  //
+  // In this function as well as many below, we collect how much time we
+  // spend here and collect that in a section called "Setup dof systems".
```
Comment (davydden) [inline examples/step-32/step-32.cc]: collect - collect, maybe rephrase a bit?

Comment (bangerth) [reply examples/step-32/step-32.cc]: Rephrased.

**Comment by masterleinad** [comment]
> /home/bob/source/examples/step-32/step-32.cc:1993:5: error: use of undeclared identifier 'OutputTimer'
    OutputTimer::Scope timing_section (computing_timer, "Setup dof systems");

That should be `TimerOutput`, isn't it?

**Comment by bangerth** [comment]
Hm, yes, that's bizarre. I thought I had tested that properly :-( Should work now!

**Comment by bangerth** [comment]
Actually: It *actually* works now. I did run the program with `mpirun -np 32` and it produces essentially the same output as the one in the stored `results.dox` file. In particular, the number of calls to each section of the `TimerOutput` object are the same.
